doc/whatsnew/v1.2.0.rst

```diff
@@ -54,9 +55,12 @@ Fixes
   (:pull_request:`660`)
 * Improve performance for Python 3 when dealing with compressed multi-frame
   Pixel Data with pillow and jpeg-ls (:issue:`682`).
-* Improve performance of bit unpacking for non-PyPy2 interpreters
-  (:pull_request:`715`)
+* Improve performance of bit unpacking (:pull_request:`715`)
 * First character set no longer removed (:issue:`707`)
 * Fixed RLE decoded data having the wrong byte order (:pull_request:`729`)
 * Fixed RLE decoded data having the wrong planar configuration
   (:pull_request:`729`)
+* Fixed numpy arrays returned by the pixel data handlers sometimes being
+  read-only. Read-only arrays are still available for uncompressed transfer
+  syntaxes via a keyword argument for the numpy pixel data handler.
+  (:issue:`717`)
```

Comment:
Since people should normally use simply `pix = ds.pixel_array` and not `get_pixeldata` directly, do we need a setting in Dataset (perhaps via dcmread?) to make this more accessible?

Reply:
...but we would have to figure out how to deal with it for other pixel data.

Also should perhaps mention the memory savings of read-only in the release notes as the advantage for bothering to set it.

Reply:
We could add another config option, or handler module option.

Reply:
I'd like to think this through a bit.  I see this as a very high-level user-friendly option, so the handler module wouldn't be my choice for specifying this, because it requires the user thinking through which handlers might be used in their set of images (which might even change over the lifetime of their script).  OTOH, they would definitely know if their code is not going to modify pixel data, but simply read and analyze it.

I think this PR has solved the issue to bring us back to pre-1.1 behavior. I'll open a new issue and we can think it through more.